Mocha testing

mayppong opened the pull request:
Here's a sample of how to write simple test cases for sockets. I've written up 2: one for setting team name and one for setting player's initials.

To test it out, run
-  `npm install` to install the test dependencies
- `npm start` to start the web server. We have to do this manually for now
- `./node_modules/.bin/mocha` to run the tests

Commit:
Create an `enable()` function for each module

Just like chat module, we create a new initialization function for each
module that the socket could call to get started.

Commit:
Write a sample test for setting new team name

Commit:
Refactor join and leave team functionality.

Commit:
Write a test for setting user initials

whiterook6 commented on lib/player.js:
```diff
@@ -45,12 +56,13 @@ var Player = {
 	// save player initials
-	update_initials: function(socket) {
+	update_initials: function(socket, initials) {
 		if (!socket || !socket.player_id || !socket.initials){
```
if you're taking in the initials, no need to check them here, or to have them set on the socket in the first place.

whiterook6 commented on lib/team.js:
```diff
@@ -1,4 +1,44 @@
 var Team = {
+	enable: function(socket) {
+		socket.on('set_team', function(data){
+			if (socket.team){
```
`````` javascript
if (socket.team && data == socket.team){
  return;
}```
``````

whiterook6 commented on package.json:
```diff
@@ -13,5 +13,9 @@
 		"sequelize": "~3.9.0",
 		"sequelize-cli": "~1.9.1"
 	},
+	"devDependencies": {
```
is this a special key for packages? Or a custom key?

mayppong replied:
The packages in devDependencies won't get install if you run `npm` in production environment.

whiterook6 commented on test/team_test.js:
```diff
@@ -0,0 +1,32 @@
+ */
+describe('Team Module', function() {
+	describe('set team', function() {
+		it('should response with new team name', function(done) {
```
should respond with a new team name

whiterook6 commented on lib/player.js:
```diff
@@ -1,4 +1,15 @@
+				Player.update_initials(socket, data);
+				Player.reset_score(socket);
+			}
+			io.sockets.emit('user.renamed', socket.initials);
```
Is this here just to support testing?

mayppong replied:
Yes and no. It's good for testing but I think it's also nice if we want the server to be able to set data as well. It's a future proof thing.

Commit:
Check if team name data is correct when setting

Commit:
No need to set current inits when setting inits

Since we're about to set a new data anyway, there's no point in checking
whethere there is an existing data or not.

Commit:
Fix spelling mistake in test
